File: 006.rb
class SumSquareDifference
  attr_reader :answer

  def initialize(range)
    @range = range
    @answer = difference
  end

  def difference
    square_of_sum - sum_of_squares
  end

  def sum_of_squares
    @range.map { |n| n**2 }.reduce(:+)
  end

  def square_of_sum
    (@range.reduce(:+))**2
  end
end

puts SumSquareDifference.new(1..10).answer
puts SumSquareDifference.new(1..100).answer

File: 006.js
var _ = require('lazy.js');

var square = function square (n) {
  return Math.pow(n, 2);
};

var sum = function sum (n, m) {
  return n + m;
};

var sumOfSquares = function sumOfSquares (range) {
  return range.map(square).reduce(sum);
};

var squareOfSum = function squareOfSum (range) {
  return square(range.reduce(sum), 2);
};

var difference = function difference (range) {
  return squareOfSum(range) - sumOfSquares(range);
};

console.log(difference(_.range(1,11)));
console.log(difference(_.range(1,101)));
